scripts/promote-beta-experimental-opt-in.ts

```diff
@@ -24,4 +24,6 @@ void runPromoteJob(jobName, async () => {
       branch: `beta-experimental-opt-in-${ampVersion}`,
     };
   });
+}).catch((error) => {
+  throw error;
```

Comment:
I'm not sure this works that way - this function will only ever run asynchronously, so it's basically like writing:
```typescript
async function func() {
  try {
    await doSomethingThatCanThrow();
  } catch (error) {
    throw error;
  }
}
```

Reply:
Soo I tested it by rejecting a promise in promote-job.ts with `reject('hi esther')`... it handles the error by printing `ERROR: hi esther`. This is what we want, right?

Reply:
I ran both of these scripts files:

```typescript
async function oopsie() {
  await Promise.reject('evil');
}

oopsie();
```

and:

```typescript
async function oopsie() {
  await Promise.reject('evil');
}

oopsie().catch((error) => {
  throw error;
});
```

As `npx ts-node ./file.ts` and in both cases received:
```
[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "evil".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```

What's the difference you're seeing between with/without the .catch call?